open-machine-learning-jupyter-book/deep-learning/image-classification.md

```diff
@@ -0,0 +1,1034 @@
+            x = Batch_Normalization(x, training=self.training, scope=scope+'_batch1')
+            x = Relu(x)
+            # x = conv_layer(x, filter=self.filters, kernel=[1,1], layer_name=scope+'_conv1')
+            
+            # https://github.com/taki0112/Densenet-Tensorflow/issues/10
+            
+            in_channel = x.shape[-1]
+            x = conv_layer(x, filter=in_channel*0.5, kernel=[1,1], layer_name=scope+'_conv1')
+            x = Drop_out(x, rate=dropout_rate, training=self.training)
+            x = Average_pooling(x, pool_size=[2,2], stride=2)
+
+            return x
+
+    def dense_block(self, input_x, nb_layers, layer_name):
+        with tf.name_scope(layer_name):
+            layers_concat = list()
+            layers_concat.append(input_x)
+
+            x = self.bottleneck_layer(input_x, scope=layer_name + '_bottleN_' + str(0))
+
+            layers_concat.append(x)
+
+            for i in range(nb_layers - 1):
+                x = Concatenation(layers_concat)
+                x = self.bottleneck_layer(x, scope=layer_name + '_bottleN_' + str(i + 1))
+                layers_concat.append(x)
+
+            x = Concatenation(layers_concat)
+
+            return x
+
+    def Dense_net(self, input_x):
+        x = conv_layer(input_x, filter=2 * self.filters, kernel=[7,7], stride=2, layer_name='conv0')
+        x = Max_Pooling(x, pool_size=[3,3], stride=2)
+
+        for i in range(self.nb_blocks) :
+            # 6 -> 12 -> 48
+            x = self.dense_block(input_x=x, nb_layers=4, layer_name='dense_'+str(i))
+            x = self.transition_layer(x, scope='trans_'+str(i))
+
+        """
+        x = self.dense_block(input_x=x, nb_layers=6, layer_name='dense_1')
+        x = self.transition_layer(x, scope='trans_1')
+        x = self.dense_block(input_x=x, nb_layers=12, layer_name='dense_2')
+        x = self.transition_layer(x, scope='trans_2')
+        x = self.dense_block(input_x=x, nb_layers=48, layer_name='dense_3')
+        x = self.transition_layer(x, scope='trans_3')
+        """
+
+        x = self.dense_block(input_x=x, nb_layers=32, layer_name='dense_final')
+
+        # 100 Layer
+        x = Batch_Normalization(x, training=self.training, scope='linear_batch')
+        x = Relu(x)
+        x = Global_Average_Pooling(x)
+        x = flatten(x)
+        x = Linear(x)
+
+        # x = tf.reshape(x, [-1, 10])
+        return x
+```
+
+### MobileNet
+
+MobileNets are based on a streamlined architecture that uses depth-wise separable convolutions to build light weight deep neural networks. We introduce two simple global hyper-parameters that efficiently trade off between latency and accuracy. These hyper-parameters allow the model builder to choose the right sized model for their application based on the constraints of the problem.
+Besides, the standard convolutional filters for normal CNNs are replaced by two layers: depthwise convolution and pointwise convolution to build a depthwise separable filter.
```

Comment:
Give a new line. If you are using vscode, please install a markdownlint plugin.

Reply:
Done, and I will find a substitution for pycharm.